renderdoc/driver/vulkan/vk_pixelhistory.cpp

```diff
@@ -1410,6 +1441,671 @@ struct TestsFailedCallback : public VulkanPixelHistoryCallback
+    colourCopyParams.srcImage = m_PerFragmentImage;
+    colourCopyParams.srcImageLayout =
+        VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;    // TODO: image layout
+    colourCopyParams.srcImageFormat = VK_FORMAT_R32G32B32A32_SFLOAT;
+    colourCopyParams.imageOffset.x = int32_t(m_X);
+    colourCopyParams.imageOffset.y = int32_t(m_Y);
+    colourCopyParams.imageOffset.z = 0;
+    colourCopyParams.dstBuffer = m_DstBuffer;
+
+    for(uint32_t f = 0; f < numFragmentsInEvent; f++)
+    {
+      for(uint32_t i = 0; i < 2; i++)
+      {
+        m_pDriver->GetCmdRenderState().graphics.pipeline = GetResID(pipesIter[i]);
+
+        m_pDriver->GetCmdRenderState().BeginRenderPassAndApplyState(
+            m_pDriver, cmd, VulkanRenderState::BindGraphics);
+
+        {
+          rdcarray<VkClearAttachment> atts(2);
+          atts[0].aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+          atts[1].aspectMask = VK_IMAGE_ASPECT_STENCIL_BIT;
+          VkClearRect rect = {};
+          rect.rect.offset.x = m_X;
+          rect.rect.offset.y = m_Y;
+          rect.rect.extent.width = 1;
+          rect.rect.extent.height = 1;
+          rect.baseArrayLayer = 0;
+          rect.layerCount = 1;
+          ObjDisp(cmd)->CmdClearAttachments(Unwrap(cmd), (uint32_t)atts.size(), atts.data(), 1,
+                                            &rect);
+        }
+
+        // Update stencil reference to the current fragment index, so that we get values
+        // for a single fragment only.
+        ObjDisp(cmd)->CmdSetStencilReference(Unwrap(cmd), VK_STENCIL_FACE_FRONT_AND_BACK, f);
+        const DrawcallDescription *drawcall = m_pDriver->GetDrawcall(eid);
+        if(drawcall->flags & DrawFlags::Indexed)
+          ObjDisp(cmd)->CmdDrawIndexed(Unwrap(cmd), drawcall->numIndices, drawcall->numInstances,
+                                       drawcall->indexOffset, drawcall->baseVertex,
+                                       drawcall->instanceOffset);
+        else
+          ObjDisp(cmd)->CmdDraw(Unwrap(cmd), drawcall->numIndices, drawcall->numInstances,
+                                drawcall->vertexOffset, drawcall->instanceOffset);
+        state.EndRenderPass(cmd);
+
+        m_pDriver->GetDebugManager()->PixelHistoryCopyPixel(
+            cmd, colourCopyParams, (fragsProcessed + f) * sizeof(PerFragmentInfo) + 16 * i);
+      }
+    }
+
+    // Restore the original framebuffer and render pass, so that we can get
+    // post modification values.
+    state.SetFramebuffer(prevState.GetFramebuffer(), prevState.GetFramebufferAttachments());
+    state.renderPass = prevState.renderPass;
+    colourCopyParams.srcImage = m_Image;
+    colourCopyParams.srcImageFormat = m_SrcFormat;
+
+    // For every fragment except the last one.
+    // The framebuffer should have the values that it had before this event.
+    // Should we use the original framebuffer? Or the new framebuffer?
+    // If we use the new one, it won't have the
```

Comment:
This comment seems cut off?

Reply:
Oops, sorry. Removed now. That was me thinking out in code comments.